### PR TITLE
Add dumps endpoint support

### PR DESCRIPTION
**Pull request by claudiunicolaa**
Implementation for /dumps https://github.com/meilisearch/meilisearch-go/issues/82

- [x] Implement dump creation route
- [x] Implement get dump status route
- [x] Add tests
- [x] Add code samples

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -99,6 +99,12 @@ type Keys struct {
 	Private string `json:"private,omitempty"`
 }
 
+// Dump indicate information about an dump
```
Comment (eskombro) [inline types.go:102]:
```suggestion
// Dump indicate information about an dump
//
// Documentation: https://docs.meilisearch.com/reference/api/dump.html
```

**Review by eskombro — APPROVED** [review]
Hello, and thanks @claudiunicolaa !!!

Thanks for taking care of this issue! 

I just reviewed it and it works perfectly as expected. GJ for the tests and the code samples :)

I just added a link to the dumps documentation in a comment, in case someone needs more info.

Can you add it please?

Otherwise it is ready to merge from my POV ! 🎉

File: types.go (hunk 1 above)
Comment (claudiunicolaa) [reply types.go:102]: Done...in the same commit but pushed with force 'cause I didn't want to "pollute" with an extra commit just for that.

Comment (eskombro) [reply types.go:102]: No problem, we squash and merge so in the end there is not much pollution, but thanks for the info! :)

**Comment by eskombro** [comment]
Bors try

File: types.go (hunk 1 above)
Comment (claudiunicolaa) [reply types.go:102]: good to know

**Comment by eskombro** [comment]
Thanks a lot @claudiunicolaa for this addition! 🎉 

Bors merge

File: types.go (hunk 1 above)
Comment (eskombro) [reply types.go:102]: Actually Bors creates a merge commit so the history looks like this:
https://github.com/meilisearch/meilisearch-go/network

**Review by eskombro — APPROVED** [review]

**Comment by eskombro** [comment]
Oh, the force push dismissed my approval! Here we go

Bors merge

**Comment by eskombro** [comment]
Oh, the force push dismissed my approval! Here we go again

Bors merge

**Comment by eskombro** [comment]
FYI @claudiunicolaa: Your commit is not signed by your GitHub user, so it doesn't appear as a contributor :(
Check your git config 👌🏽

**Pull request merged**